### PR TITLE
feat(StorybookBlock): validate url + simplify block

### PR DESCRIPTION
**Pull request by celine-s**
Ticket: https://app.clickup.com/t/2qp0v45

design: approved from Martin

### CODE REVIEW

**Review by rawi96 — APPROVED** [review]

File: packages/storybook-block/src/StorybookBlock.tsx [hunk 1]
```diff
@@ -59,40 +79,39 @@ export const StorybookBlock: FC<BlockProps> = ({ appBridge }) => {
...
         setBlockSettings({
             ...blockSettings,
-            url: localUrl,
+            url: addHttps(localUrl),
         });
     };
 
     useEffect(() => {
+        if (!url) {
+            setLocalUrl('');
```
Comment (triggertoo) [inline packages/storybook-block/src/StorybookBlock.tsx:88]: imho this should not happen within a `useEffect`

File: packages/storybook-block/src/StorybookBlock.tsx [hunk 2]
```diff
@@ -59,40 +79,39 @@ export const StorybookBlock: FC<BlockProps> = ({ appBridge }) => {
...
+        }
         if (url !== '') {
             setIsReadyForPrint(false);
-            const newIframeUrl = new URL(url);
-            newIframeUrl.searchParams.set('nav', 'false');
+        } else if (url === '') {
```
Comment (triggertoo) [inline packages/storybook-block/src/StorybookBlock.tsx:92]:
```suggestion
        } else if (url.trim().length === 0) {
```

File: packages/storybook-block/src/StorybookBlock.tsx [hunk 3]
```diff
@@ -59,40 +79,39 @@ export const StorybookBlock: FC<BlockProps> = ({ appBridge }) => {
...
 
-            newIframeUrl.searchParams.set('panel', panelValue);
+        const hasAddons = style === StorybookStyle.Default;
+        const includesIframe = iframeUrl?.pathname.toString().includes('iframe.html');
+        const positionValue = positioning === StorybookPosition.Horizontal ? 'right' : 'bottom';
 
-            if (shouldAddIframeToUrl && !includesIframe) {
-                newIframeUrl.pathname = `${newIframeUrl.pathname}iframe.html`;
-            }
+        iframeUrl?.searchParams.set('panel', !hasAddons ? 'false' : positionValue);
```
Comment (triggertoo) [inline packages/storybook-block/src/StorybookBlock.tsx:104]:
```suggestion
        iframeUrl?.searchParams.set('panel', hasAddons ? positionValue : 'false');
```

File: packages/storybook-block/src/StorybookBlock.tsx [hunk 4]
```diff
@@ -47,58 +44,40 @@ export const StorybookBlock: FC<BlockProps> = ({ appBridge }) => {
...
-
-            setIframeUrl(newIframeUrl);
-        } else if (url === '') {
-            setIsReadyForPrint(true);
-            deleteUrl();
-        }
-        // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, [url, style, positioning]);
+        setStorybookUrl(url);
+        setIsReadyForPrint(url !== '');
```
Comment (rawi96) [inline packages/storybook-block/src/StorybookBlock.tsx:63]: Block should initially be ready for print.
As soon as a link is entered it should NOT be ready.
And as soon as the iframe is loaded (or the URL is invalid) it should be ready again.

Currently the block is NOT ready when no URL is entered.

**Review by rawi96 — APPROVED** [review]

**Review by Walkeezy — APPROVED** [review]
💅🏻 nice!

**Pull request merged**